samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java

```diff
@@ -630,11 +601,58 @@ public void validateStream(StreamSpec streamSpec) throws StreamValidationExcepti
   @Override
   public void deleteMessages(Map<SystemStreamPartition, String> offsets) {
     if (deleteCommittedMessages) {
+      if (adminClient == null) {
+        adminClient = AdminClient.create(createAdminClientProperties());
+      }
       KafkaSystemAdminUtilsScala.deleteMessages(adminClient, offsets);
       deleteMessageCalled = true;
     }
   }
 
+  private Properties createAdminClientProperties() {
+    // populate brokerList from either consumer or producer configs
+    Properties props = new Properties();
+    // included SSL settings if needed
```

Comment:
Do we want to make this method protected instead and have the extenders determine how they want to generate properties?

Reply:
yes. Will change.